parsing/parser.mly

```diff
@@ -3388,10 +3388,13 @@ type_variance:
   | INFIXOP2
       { if $1 = "+!" then Covariant, Injective else
         if $1 = "-!" then Contravariant, Injective else
+        if $1 = "+-" then Bivariant, NoInjectivity else
+        if $1 = "+-!" then Bivariant, Injective else
```

Comment:
Do we have a good reason and mnemonic to allow only `+-` and not `-+`?

Reply:
I thought it would be better to minimize the change to the syntax, and did not think that the symmetry here could help a lot of users.  Do you have other opinions?

Reply:
I would tend to allow any permutation of `±` and `!`, and to consider that `±` can be written either `+-` or `-+`.
In other words, adding `+-`, `-+`, `+-!`, `-+!`, `!+-`, and `!-+` (but not `+!-` nor `-!+`).

Reply:
Okay, my preference was anyway not strong, and I am happy to follow your suggestions. I will add `-+` and `-+!`. 
(Btw, at the first attempt Jacques and I considered `+ -!` `+- !`, etc. too, and soon realized that the list will become huge :)